otter/metrics.py

```diff
@@ -253,6 +254,40 @@ def connect_cass_servers(reactor, config):
         seed_endpoints, config['keyspace'], disconnect_on_cancel=True)
 
 
+def log_divergent_groups(clock, divergent_groups, log, timeout, group_metrics):
+    """
+    Log groups that have not changed and been divergent for long time
+    """
+    dg = divergent_groups
+    converged, diverged = partition_bool(
+        lambda gm: gm.actual + gm.pending == gm.desired, group_metrics)
+    # stop tracking all converged groups
+    for gm in converged:
+        dg.pop((gm.tenant_id, gm.group_id), None)
+    # Start tracking divergent groups depending on whether they've changed
+    now = clock.seconds()
+    for gm in diverged:
+        pair = (gm.tenant_id, gm.group_id)
+        if pair in dg:
+            last_time, values = dg[pair]
+            if values != hash((gm.desired, gm.actual, gm.pending)):
+                del dg[pair]
```

Comment:
Right, but the time it has been stuck should start when we first see that particular combination of `(gm.desired, gm.actual, gm.pending)`, it seems.